### PR TITLE
Module cleanup

### PR DESCRIPTION
**Pull request by amitaibu**
Still WIP, and completely un-tested ;)
- [ ] Make methods similar to RESTful
- [ ] Remove preprocess -- it's not up for this module to change the field value. What was the original use case?
- [ ] Pass more arguments to the validate callbacks, and allow passing sub_property (e.g. $wrapper->body->value) instead of the whole value array.

@RoySegall @mateu-aguilo-bosch
Next is some cleanup for the name property/ field_name -- although I'm not sure which one is the correct one

### CODE REVIEW

**Comment by RoySegall** [comment]
@amitaibu We also talked about that we need to check multiple fields and their values when empty.

**Comment by RoySegall** [comment]
@amitaibu It seems that  the tests are passed. I would like to have a better look on that.

**Comment by amitaibu** [comment]
More tests are probably needed as-well.

**Comment by RoySegall** [comment]
We should add multiple field and validate it as well.

**Comment by RoySegall** [comment]
@amitaibu The tests passed.

File: includes/EntityValidateBase.php [hunk 1]
```diff
@@ -366,4 +388,24 @@ public function validateFileExtension($field_name, $value) {
       $this->setError($field_name, 'The file (@file-name) extension (@extension) did not match the allowed extensions: @extensions', $params);
     }
   }
+
+  /**
+   * Validate population of the multiple field with info.
```
Comment (amitaibu) [inline includes/EntityValidateBase.php]: Validate a field with multiple cardinality is not empty.

File: includes/EntityValidateBase.php [hunk 2]
```diff
@@ -366,4 +388,24 @@ public function validateFileExtension($field_name, $value) {
...
+   *   The value of the field.
+   * @param EntityMetadataWrapper $wrapper
+   *   The wrapped entity.
+   * @param EntityMetadataWrapper $property_wrapper
+   *   The wrapped property.
+   */
+  public function validateMultipleFieldNotEmpty($field_name, $value, EntityMetadataWrapper $wrapper, EntityMetadataWrapper $property_wrapper) {
+    foreach ($property_wrapper as $delta => $sub_wrapper) {
+      if (!$sub_wrapper->value()) {
+        $this->setError($field_name, 'The delta @delta cant be empty', array('@delta' => $delta));
```
Comment (amitaibu) [inline includes/EntityValidateBase.php]: - cant => cannot
- missing dot

**Comment by RoySegall** [comment]
@amitaibu I commit the fixes. Merge?

**Comment by amitaibu** [comment]
Tests are failing.

**Comment by RoySegall** [comment]
Merged manually.

**Pull request closed without merging**